### ISSUE TITLE
Crash with Principia when clicking "finalize"

### ISSUE THREAD

**Issue by Astavie** [taniwha/Extraplanetary-Launchpads#152]
Hi!

Love the mod, but it keeps crashing when I try to finalize a ship. Looking at stderr.log it seems it is incompatible with Principia, maybe? [stderr.log](https://github.com/taniwha/Extraplanetary-Launchpads/files/3496590/stderr.log)

I also have a save file if it helps: [default.zip](https://github.com/taniwha/Extraplanetary-Launchpads/files/3496591/default.zip). I installed a few mods but for the vessel on the launchpad only [Kerbal Planetary Base Systems](https://forum.kerbalspaceprogram.com/index.php?/topic/133606-17x-kerbal-planetary-base-systems-v169-25-april-2019/), Principia and Extraplanetary Launchpads is needed.

**Comment by Mogball** [comment]
I'm also experiencing this issue. Would like to bump!

Related issue: https://github.com/taniwha/Extraplanetary-Launchpads/issues/137

I'm not sure whether this is more appropriate on the Principia Github or here, but the relevant log is

```
Log file created at: 2019/12/28 16:43:45
Running on machine: 
Log line format: [IWEF]mmdd hh:mm:ss.uuuuuu threadid file:line] msg
    @   00007FFAF2BD633A  	google::LogMessageFatal::~LogMessageFatal [0x00007FFAF2BD6339+57]
    @   00007FFAAD55933C  	principia__VesselVelocity [0x00007FFAAD55933B+673067]
    @   00007FFAAD57183F  	principia__VesselVelocity [0x00007FFAAD57183E+772654]
    @   00007FFAAD498365  	principia__InsertUnloadedPart [0x00007FFAAD498364+564]
    @   00000000206AFFEC  	(No symbol) [0x00000000206AFFEB]
F1228 16:43:45.887115 15564 plugin.cpp:1499] Check failed: inserted part_id: 0
```

**Comment by Mogball** [comment]
It would seem that EPL is not giving the new vessel unique part IDs with respect to the new vessel docked to the launch pad and the rest of the vessel, which causes Principia fail a map insertion.

**Comment by taniwha** [comment]
EL does not give any IDs to anything: all ID assignment is done by KSP. EL creates the new vessel as a separate vessel then docks it to the parent vessel using KSP's docking code.

**Comment by Senaurus** [comment]
I have same symptoms but I do not have Principia installed. So the issue might not be related to Principia. And I belive long time ago I had it installed and I would get that error, so I uninstalled it, and still would get the same game crash problem but without the error. So again it is most likely not Principia.
Some more info:
1. This does not happen all the time. My success rate depends on how complex the surrounding is. for example the more parts I have on the station to which construction dock is attached to, the more likely the crash will happen. 
2. up until recently I was running the game on KSP 1.4.7 with 6.2.2 of EL, which also had this issue, and one of the that was happening is that the newly spanned parts would span not next to the construction dock but with construction dock inside. So looks like size/displacement calculation was off. I resolved that issue by attaching very long I-beams or girders to it in such a way that I knew that there would be no overlap. That would reduce the chance of freezing by order of magnitude. So instead of crashing something 4 out of 5 times, it would crash only 1 out 5 times. 

I think the problem might be with KSP checking for collisions and that is where it dies, since it does not really expect for something appearing out of nowhere, which is what this mod does.

P.S.: So I'm running 1.8.1 KSP right now with 6.7.2 of EL (and a ton of other mods.) and just verified my #2. I un-docked Assembly plant to which I have the construction port attached moved it about 50 meters away form my station, and finalization was successful this time, but before that I tried without undocking, and with undocking but not moving away, other than just drifting a little bit, and both times the game crashed. This really makes me think that it has to do with some kind of collision check. Also it looks like the sizing/displacement calculations are correct on this version of the EL. The spawned parts were position correctly with out any overlap.
